dataplex/snippets/src/main/java/dataplex/SearchEntries.java

```diff
@@ -0,0 +1,72 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dataplex;
+
+// [START dataplex_search_entries]
+import com.google.cloud.dataplex.v1.CatalogServiceClient;
+import com.google.cloud.dataplex.v1.Entry;
+import com.google.cloud.dataplex.v1.SearchEntriesRequest;
+import com.google.cloud.dataplex.v1.SearchEntriesResult;
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class SearchEntries {
+
+  public static void main(String[] args) throws IOException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "MY_PROJECT_ID";
+    // How to write query for search: https://cloud.google.com/dataplex/docs/search-syntax
+    String query = "MY_QUERY";
+
+    List<Entry> entries = searchEntries(projectId, query);
+    entries.forEach(entry -> System.out.println("Entry name found in search: " + entry.getName()));
+  }
+
+  // Method to search Entries located in projectId matching query
+  public static List<Entry> searchEntries(String projectId, String query) throws IOException {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests.
+    try (CatalogServiceClient client = CatalogServiceClient.create()) {
+      SearchEntriesRequest searchEntriesRequest =
+          SearchEntriesRequest.newBuilder()
+              // There are 2 ways to limit the scope of the search:
+              // 1) Set the "scope" field, with either "projects/<>" or "organizations/<>",
+              // which will limit the search to resources residing in given project / organization.
+              // 2) Leave "scope" field blank. In that case scope will be limited to organization in
+              // which project from "name" field is located.
+              // Please note that field "name" must always be filled, as the request itself will be
+              // attributed to that.
+              .setName(String.format("projects/%s/locations/global", projectId)) // required
+              .setScope(String.format("projects/%s", projectId)) // optional
+              .setQuery(query)
+              .build();
+
+      CatalogServiceClient.SearchEntriesPagedResponse searchEntriesResponse =
+          client.searchEntries(searchEntriesRequest);
+      // Paging is implicitly handled by .iterateAll(), all results will be returned
+      List<SearchEntriesResult> searchEntriesResults =
+          ImmutableList.copyOf(searchEntriesResponse.iterateAll());
```

Comment:
I'd suggest to remove the `iterateAll` from this search implementation. This function will lazily paginate over all available data and for organizations with a lot of data might results in a large exectution time.

Reply:
Resolve the conversation as changes were applied.